<h2>Live Coding</h2>

In [1]:
import pandas as pd

# Leer un archivo CSV
df_grd = pd.read_csv("data/chunk1.csv", dtype=str, low_memory=False)
df_grd.dtypes

COD_HOSPITAL           object
CIP_ENCRIPTADO         object
SEXO                   object
FECHA_NACIMIENTO       object
ETNIA                  object
                        ...  
IR_29301_COD_GRD       object
IR_29301_PESO          object
IR_29301_SEVERIDAD     object
IR_29301_MORTALIDAD    object
HOSPPROCEDENCIA        object
Length: 129, dtype: object

In [5]:
df_grd.columns

Index(['COD_HOSPITAL', 'CIP_ENCRIPTADO', 'SEXO', 'FECHA_NACIMIENTO', 'ETNIA',
       'PROVINCIA', 'COMUNA', 'NACIONALIDAD', 'PREVISION', 'SERVICIO_SALUD',
       ...
       'FECHAPROCEDIMIENTO1', 'FECHAINTERV1', 'ESPECIALIDADINTERVENCION',
       'MEDICOALTA_ENCRIPTADO', 'USOSPABELLON', 'IR_29301_COD_GRD',
       'IR_29301_PESO', 'IR_29301_SEVERIDAD', 'IR_29301_MORTALIDAD',
       'HOSPPROCEDENCIA'],
      dtype='object', length=129)

<h3>1.Eliminar columnas innecesarias.</h3>

In [2]:
# 1) identificar columnas tipo fecha (por nombre)
cols_fecha = [c for c in df_grd.columns if c.upper().startswith("FECHA")]
cols_fecha

['FECHA_NACIMIENTO',
 'FECHA_INGRESO',
 'FECHATRASLADO1',
 'FECHATRASLADO2',
 'FECHATRASLADO3',
 'FECHATRASLADO4',
 'FECHATRASLADO5',
 'FECHATRASLADO6',
 'FECHATRASLADO7',
 'FECHATRASLADO8',
 'FECHATRASLADO9',
 'FECHAALTA',
 'FECHAPROCEDIMIENTO1',
 'FECHAINTERV1']

In [12]:
# 2) dejar solo FECHA_NACIMIENTO dentro de esas
cols_a_borrar = [c for c in cols_fecha if (c != "FECHA_NACIMIENTO" and c != "FECHA_INGRESO")]

# 3) borrar
df_sin_fechas = df_grd.drop(columns=cols_a_borrar)

df_sin_fechas.shape, len(cols_a_borrar), cols_a_borrar[:10]


((10000, 117),
 12,
 ['FECHATRASLADO1',
  'FECHATRASLADO2',
  'FECHATRASLADO3',
  'FECHATRASLADO4',
  'FECHATRASLADO5',
  'FECHATRASLADO6',
  'FECHATRASLADO7',
  'FECHATRASLADO8',
  'FECHATRASLADO9',
  'FECHAALTA'])

In [16]:
df_sin_fechas.columns

Index(['COD_HOSPITAL', 'CIP_ENCRIPTADO', 'SEXO', 'FECHA_NACIMIENTO', 'ETNIA',
       'PROVINCIA', 'COMUNA', 'NACIONALIDAD', 'PREVISION', 'SERVICIO_SALUD',
       ...
       'PROCEDIMIENTO30', 'MEDICOINTERV1_ENCRIPTADO',
       'ESPECIALIDADINTERVENCION', 'MEDICOALTA_ENCRIPTADO', 'USOSPABELLON',
       'IR_29301_COD_GRD', 'IR_29301_PESO', 'IR_29301_SEVERIDAD',
       'IR_29301_MORTALIDAD', 'HOSPPROCEDENCIA'],
      dtype='object', length=117)

In [17]:
cols_a_borrar = [c for c in df_sin_fechas.columns if "PROCEDIMIENTO" in c]

In [18]:
cols_a_borrar

['PROCEDIMIENTO1',
 'PROCEDIMIENTO2',
 'PROCEDIMIENTO3',
 'PROCEDIMIENTO4',
 'PROCEDIMIENTO5',
 'PROCEDIMIENTO6',
 'PROCEDIMIENTO7',
 'PROCEDIMIENTO8',
 'PROCEDIMIENTO9',
 'PROCEDIMIENTO10',
 'PROCEDIMIENTO11',
 'PROCEDIMIENTO12',
 'PROCEDIMIENTO13',
 'PROCEDIMIENTO14',
 'PROCEDIMIENTO15',
 'PROCEDIMIENTO16',
 'PROCEDIMIENTO17',
 'PROCEDIMIENTO18',
 'PROCEDIMIENTO19',
 'PROCEDIMIENTO20',
 'PROCEDIMIENTO21',
 'PROCEDIMIENTO22',
 'PROCEDIMIENTO23',
 'PROCEDIMIENTO24',
 'PROCEDIMIENTO25',
 'PROCEDIMIENTO26',
 'PROCEDIMIENTO27',
 'PROCEDIMIENTO28',
 'PROCEDIMIENTO29',
 'PROCEDIMIENTO30']

In [19]:
df_sin_fechas = df_sin_fechas.drop(columns=cols_a_borrar)

len(cols_a_borrar), cols_a_borrar[:10]

(30,
 ['PROCEDIMIENTO1',
  'PROCEDIMIENTO2',
  'PROCEDIMIENTO3',
  'PROCEDIMIENTO4',
  'PROCEDIMIENTO5',
  'PROCEDIMIENTO6',
  'PROCEDIMIENTO7',
  'PROCEDIMIENTO8',
  'PROCEDIMIENTO9',
  'PROCEDIMIENTO10'])

In [20]:
df_sin_fechas.columns

Index(['COD_HOSPITAL', 'CIP_ENCRIPTADO', 'SEXO', 'FECHA_NACIMIENTO', 'ETNIA',
       'PROVINCIA', 'COMUNA', 'NACIONALIDAD', 'PREVISION', 'SERVICIO_SALUD',
       'TIPO_PROCEDENCIA', 'TIPO_INGRESO', 'ESPECIALIDAD_MEDICA',
       'TIPO_ACTIVIDAD', 'FECHA_INGRESO', 'SERVICIOINGRESO',
       'SERVICIOTRASLADO1', 'SERVICIOTRASLADO2', 'SERVICIOTRASLADO3',
       'SERVICIOTRASLADO4', 'SERVICIOTRASLADO5', 'SERVICIOTRASLADO6',
       'SERVICIOTRASLADO7', 'SERVICIOTRASLADO8', 'SERVICIOTRASLADO9',
       'SERVICIOALTA', 'TIPOALTA', 'CONDICIONDEALTANEONATO1', 'PESORN1',
       'SEXORN1', 'RN1ESTADO', 'CONDICIONDEALTANEONATO2', 'PESORN2', 'SEXORN2',
       'RN2ESTADO', 'CONDICIONDEALTANEONATO3', 'PESORN3', 'SEXORN3',
       'RN3ESTADO', 'CONDICIONDEALTANEONATO4', 'PESORN4', 'SEXORN4',
       'RN4ESTADO', 'DIAGNOSTICO1', 'DIAGNOSTICO2', 'DIAGNOSTICO3',
       'DIAGNOSTICO4', 'DIAGNOSTICO5', 'DIAGNOSTICO6', 'DIAGNOSTICO7',
       'DIAGNOSTICO8', 'DIAGNOSTICO9', 'DIAGNOSTICO10', 'DIAGNOSTICO11',
     

In [21]:
cols_a_borrar = [c for c in df_sin_fechas.columns if "DIAGNOSTICO" in c]
df_sin_fechas = df_sin_fechas.drop(columns=cols_a_borrar)

len(cols_a_borrar), cols_a_borrar[:10]

(35,
 ['DIAGNOSTICO1',
  'DIAGNOSTICO2',
  'DIAGNOSTICO3',
  'DIAGNOSTICO4',
  'DIAGNOSTICO5',
  'DIAGNOSTICO6',
  'DIAGNOSTICO7',
  'DIAGNOSTICO8',
  'DIAGNOSTICO9',
  'DIAGNOSTICO10'])

In [22]:
df_sin_fechas.columns

Index(['COD_HOSPITAL', 'CIP_ENCRIPTADO', 'SEXO', 'FECHA_NACIMIENTO', 'ETNIA',
       'PROVINCIA', 'COMUNA', 'NACIONALIDAD', 'PREVISION', 'SERVICIO_SALUD',
       'TIPO_PROCEDENCIA', 'TIPO_INGRESO', 'ESPECIALIDAD_MEDICA',
       'TIPO_ACTIVIDAD', 'FECHA_INGRESO', 'SERVICIOINGRESO',
       'SERVICIOTRASLADO1', 'SERVICIOTRASLADO2', 'SERVICIOTRASLADO3',
       'SERVICIOTRASLADO4', 'SERVICIOTRASLADO5', 'SERVICIOTRASLADO6',
       'SERVICIOTRASLADO7', 'SERVICIOTRASLADO8', 'SERVICIOTRASLADO9',
       'SERVICIOALTA', 'TIPOALTA', 'CONDICIONDEALTANEONATO1', 'PESORN1',
       'SEXORN1', 'RN1ESTADO', 'CONDICIONDEALTANEONATO2', 'PESORN2', 'SEXORN2',
       'RN2ESTADO', 'CONDICIONDEALTANEONATO3', 'PESORN3', 'SEXORN3',
       'RN3ESTADO', 'CONDICIONDEALTANEONATO4', 'PESORN4', 'SEXORN4',
       'RN4ESTADO', 'MEDICOINTERV1_ENCRIPTADO', 'ESPECIALIDADINTERVENCION',
       'MEDICOALTA_ENCRIPTADO', 'USOSPABELLON', 'IR_29301_COD_GRD',
       'IR_29301_PESO', 'IR_29301_SEVERIDAD', 'IR_29301_MORTALIDAD',
    

In [23]:
cols_a_borrar = [c for c in df_sin_fechas.columns if "SERVICIOTRASLADO" in c]
df_sin_fechas = df_sin_fechas.drop(columns=cols_a_borrar)

len(cols_a_borrar), cols_a_borrar[:10]

(9,
 ['SERVICIOTRASLADO1',
  'SERVICIOTRASLADO2',
  'SERVICIOTRASLADO3',
  'SERVICIOTRASLADO4',
  'SERVICIOTRASLADO5',
  'SERVICIOTRASLADO6',
  'SERVICIOTRASLADO7',
  'SERVICIOTRASLADO8',
  'SERVICIOTRASLADO9'])

In [24]:
df_sin_fechas.columns

Index(['COD_HOSPITAL', 'CIP_ENCRIPTADO', 'SEXO', 'FECHA_NACIMIENTO', 'ETNIA',
       'PROVINCIA', 'COMUNA', 'NACIONALIDAD', 'PREVISION', 'SERVICIO_SALUD',
       'TIPO_PROCEDENCIA', 'TIPO_INGRESO', 'ESPECIALIDAD_MEDICA',
       'TIPO_ACTIVIDAD', 'FECHA_INGRESO', 'SERVICIOINGRESO', 'SERVICIOALTA',
       'TIPOALTA', 'CONDICIONDEALTANEONATO1', 'PESORN1', 'SEXORN1',
       'RN1ESTADO', 'CONDICIONDEALTANEONATO2', 'PESORN2', 'SEXORN2',
       'RN2ESTADO', 'CONDICIONDEALTANEONATO3', 'PESORN3', 'SEXORN3',
       'RN3ESTADO', 'CONDICIONDEALTANEONATO4', 'PESORN4', 'SEXORN4',
       'RN4ESTADO', 'MEDICOINTERV1_ENCRIPTADO', 'ESPECIALIDADINTERVENCION',
       'MEDICOALTA_ENCRIPTADO', 'USOSPABELLON', 'IR_29301_COD_GRD',
       'IR_29301_PESO', 'IR_29301_SEVERIDAD', 'IR_29301_MORTALIDAD',
       'HOSPPROCEDENCIA'],
      dtype='object')

In [25]:
cols_a_borrar = [c for c in df_sin_fechas.columns if "CONDICIONDEALTANEONAT" in c]
df_sin_fechas = df_sin_fechas.drop(columns=cols_a_borrar)

len(cols_a_borrar), cols_a_borrar[:10]

(4,
 ['CONDICIONDEALTANEONATO1',
  'CONDICIONDEALTANEONATO2',
  'CONDICIONDEALTANEONATO3',
  'CONDICIONDEALTANEONATO4'])

In [26]:
cols_a_borrar = [c for c in df_sin_fechas.columns if "PESORN" in c]
df_sin_fechas = df_sin_fechas.drop(columns=cols_a_borrar)

len(cols_a_borrar), cols_a_borrar[:10]

(4, ['PESORN1', 'PESORN2', 'PESORN3', 'PESORN4'])

In [27]:
cols_a_borrar = [c for c in df_sin_fechas.columns if "SEXORN" in c]
df_sin_fechas = df_sin_fechas.drop(columns=cols_a_borrar)

len(cols_a_borrar), cols_a_borrar[:10]

(4, ['SEXORN1', 'SEXORN2', 'SEXORN3', 'SEXORN4'])

In [34]:
cols_a_borrar = [c for c in df_sin_fechas.columns if "RN4ESTADO" in c]
df_sin_fechas = df_sin_fechas.drop(columns=cols_a_borrar)

len(cols_a_borrar), cols_a_borrar[:10]

(1, ['RN4ESTADO'])

In [35]:
df_sin_fechas.columns

Index(['COD_HOSPITAL', 'CIP_ENCRIPTADO', 'SEXO', 'FECHA_NACIMIENTO', 'ETNIA',
       'PROVINCIA', 'COMUNA', 'NACIONALIDAD', 'PREVISION', 'SERVICIO_SALUD',
       'TIPO_PROCEDENCIA', 'TIPO_INGRESO', 'ESPECIALIDAD_MEDICA',
       'TIPO_ACTIVIDAD', 'FECHA_INGRESO', 'SERVICIOINGRESO', 'SERVICIOALTA',
       'TIPOALTA', 'MEDICOINTERV1_ENCRIPTADO', 'ESPECIALIDADINTERVENCION',
       'MEDICOALTA_ENCRIPTADO', 'USOSPABELLON', 'IR_29301_COD_GRD',
       'IR_29301_PESO', 'IR_29301_SEVERIDAD', 'IR_29301_MORTALIDAD',
       'HOSPPROCEDENCIA'],
      dtype='object')

<h3>2. Convertir columnas de string a datetime.</h3>

In [36]:
df_sin_fechas["FECHA_NACIMIENTO"].value_counts()

FECHA_NACIMIENTO
2005-06-24    9
2019-04-01    6
2003-03-13    6
2019-10-25    5
2019-07-22    5
             ..
2017-03-05    1
1960-02-28    1
1947-09-15    1
1956-02-07    1
1954-05-27    1
Name: count, Length: 8358, dtype: int64

In [37]:
df_sin_fechas["FECHA_NACIMIENTO"].dtype


dtype('O')

In [ ]:
s = df_sin_fechas["FECHA_NACIMIENTO"].astype("string").str.strip()
s = s.str.replace(".", "/", regex=False).str.replace("-", "/", regex=False)

df_sin_fechas["FECHA_NACIMIENTO"] = pd.to_datetime(
    s,
    errors="coerce"
)

df_sin_fechas["FECHA_NACIMIENTO"].dtype


dtype('<M8[ns]')

In [40]:
df_sin_fechas.dtypes

COD_HOSPITAL                        object
CIP_ENCRIPTADO                      object
SEXO                                object
FECHA_NACIMIENTO            datetime64[ns]
ETNIA                               object
PROVINCIA                           object
COMUNA                              object
NACIONALIDAD                        object
PREVISION                           object
SERVICIO_SALUD                      object
TIPO_PROCEDENCIA                    object
TIPO_INGRESO                        object
ESPECIALIDAD_MEDICA                 object
TIPO_ACTIVIDAD                      object
FECHA_INGRESO                       object
SERVICIOINGRESO                     object
SERVICIOALTA                        object
TIPOALTA                            object
MEDICOINTERV1_ENCRIPTADO            object
ESPECIALIDADINTERVENCION            object
MEDICOALTA_ENCRIPTADO               object
USOSPABELLON                        object
IR_29301_COD_GRD                    object
IR_29301_PE

In [41]:
s = df_sin_fechas["FECHA_INGRESO"].astype("string").str.strip()
s = s.str.replace(".", "/", regex=False).str.replace("-", "/", regex=False)

df_sin_fechas["FECHA_INGRESO"] = pd.to_datetime(
    s,
    errors="coerce"
)

df_sin_fechas["FECHA_INGRESO"].dtype

dtype('<M8[ns]')

In [42]:
df_sin_fechas.dtypes

COD_HOSPITAL                        object
CIP_ENCRIPTADO                      object
SEXO                                object
FECHA_NACIMIENTO            datetime64[ns]
ETNIA                               object
PROVINCIA                           object
COMUNA                              object
NACIONALIDAD                        object
PREVISION                           object
SERVICIO_SALUD                      object
TIPO_PROCEDENCIA                    object
TIPO_INGRESO                        object
ESPECIALIDAD_MEDICA                 object
TIPO_ACTIVIDAD                      object
FECHA_INGRESO               datetime64[ns]
SERVICIOINGRESO                     object
SERVICIOALTA                        object
TIPOALTA                            object
MEDICOINTERV1_ENCRIPTADO            object
ESPECIALIDADINTERVENCION            object
MEDICOALTA_ENCRIPTADO               object
USOSPABELLON                        object
IR_29301_COD_GRD                    object
IR_29301_PE

<h3> 3. Aplicar .set_index() y .reset_index().</h3>

In [43]:
df_sin_fechas.index


RangeIndex(start=0, stop=10000, step=1)

In [44]:
df_sin_fechas = df_sin_fechas.rename(
    columns={"CIP_ENCRIPTADO": "ID_PACIENTE"}
)


In [45]:
df_idx = df_sin_fechas.set_index("ID_PACIENTE")
df_idx.head()


,COD_HOSPITAL,SEXO,FECHA_NACIMIENTO,ETNIA,PROVINCIA,COMUNA,NACIONALIDAD,PREVISION,SERVICIO_SALUD,TIPO_PROCEDENCIA,...,TIPOALTA,MEDICOINTERV1_ENCRIPTADO,ESPECIALIDADINTERVENCION,MEDICOALTA_ENCRIPTADO,USOSPABELLON,IR_29301_COD_GRD,IR_29301_PESO,IR_29301_SEVERIDAD,IR_29301_MORTALIDAD,HOSPPROCEDENCIA
ID_PACIENTE,,,,,,,,,,,,,,,,,,,,,
1314867,118100,MUJER,1988-06-19,OTRO,CONCEPCION,SAN PEDRO DE LA PAZ,CHILE,FONASA INSTITUCIONAL - (MAI) A,CONCEPCIÓN,"CENTRO ESPECIALIDADES (CDT, CRS, CONSULTORIO A...",...,DOMICILIO,12498.0,OBSTETRICIA Y GINECOLOGÍA,2932.0,1.0,146102,"0,5744",2,1,NaN
1354418,118100,MUJER,1988-03-27,OTRO,CONCEPCION,SAN PEDRO DE LA PAZ,CHILE,FONASA INSTITUCIONAL - (MAI) D,CONCEPCIÓN,SERVICIO EMERGENCIA (DOMICILIO),...,DOMICILIO,NaN,NaN,2932.0,NaN,134161,"0,2951",1,1,NaN
239861,118100,MUJER,2017-06-17,OTRO,CONCEPCION,CONCEPCIÓN,CHILE,FONASA INSTITUCIONAL - (MAI) B,CONCEPCIÓN,SERVICIO EMERGENCIA (DOMICILIO),...,DOMICILIO,NaN,NaN,2932.0,NaN,64182,"0,6736",2,2,NaN
330326,118100,HOMBRE,2017-12-04,OTRO,CONCEPCION,CONCEPCIÓN,CHILE,FONASA INSTITUCIONAL - (MAI) A,CONCEPCIÓN,SERVICIO EMERGENCIA (DOMICILIO),...,DOMICILIO,NaN,NaN,2932.0,NaN,64172,"0,5475",2,2,NaN
1369293,118100,MUJER,1995-12-06,OTRO,CONCEPCION,FLORIDA,CHILE,FONASA INSTITUCIONAL - (MAI) C,CONCEPCIÓN,SERVICIO EMERGENCIA (DOMICILIO),...,DOMICILIO,NaN,NaN,2932.0,NaN,134141,"0,3107",1,1,NaN


In [46]:
df_back = df_idx.reset_index()
df_back.head()


,ID_PACIENTE,COD_HOSPITAL,SEXO,FECHA_NACIMIENTO,ETNIA,PROVINCIA,COMUNA,NACIONALIDAD,PREVISION,SERVICIO_SALUD,...,TIPOALTA,MEDICOINTERV1_ENCRIPTADO,ESPECIALIDADINTERVENCION,MEDICOALTA_ENCRIPTADO,USOSPABELLON,IR_29301_COD_GRD,IR_29301_PESO,IR_29301_SEVERIDAD,IR_29301_MORTALIDAD,HOSPPROCEDENCIA
0,1314867,118100,MUJER,1988-06-19,OTRO,CONCEPCION,SAN PEDRO DE LA PAZ,CHILE,FONASA INSTITUCIONAL - (MAI) A,CONCEPCIÓN,...,DOMICILIO,12498.0,OBSTETRICIA Y GINECOLOGÍA,2932.0,1.0,146102,"0,5744",2,1,NaN
1,1354418,118100,MUJER,1988-03-27,OTRO,CONCEPCION,SAN PEDRO DE LA PAZ,CHILE,FONASA INSTITUCIONAL - (MAI) D,CONCEPCIÓN,...,DOMICILIO,NaN,NaN,2932.0,NaN,134161,"0,2951",1,1,NaN
2,239861,118100,MUJER,2017-06-17,OTRO,CONCEPCION,CONCEPCIÓN,CHILE,FONASA INSTITUCIONAL - (MAI) B,CONCEPCIÓN,...,DOMICILIO,NaN,NaN,2932.0,NaN,64182,"0,6736",2,2,NaN
3,330326,118100,HOMBRE,2017-12-04,OTRO,CONCEPCION,CONCEPCIÓN,CHILE,FONASA INSTITUCIONAL - (MAI) A,CONCEPCIÓN,...,DOMICILIO,NaN,NaN,2932.0,NaN,64172,"0,5475",2,2,NaN
4,1369293,118100,MUJER,1995-12-06,OTRO,CONCEPCION,FLORIDA,CHILE,FONASA INSTITUCIONAL - (MAI) C,CONCEPCIÓN,...,DOMICILIO,NaN,NaN,2932.0,NaN,134141,"0,3107",1,1,NaN


In [58]:
df_sin_fechas[df_sin_fechas["ID_PACIENTE"] == "11208"]


,COD_HOSPITAL,ID_PACIENTE,SEXO,FECHA_NACIMIENTO,ETNIA,PROVINCIA,COMUNA,NACIONALIDAD,PREVISION,SERVICIO_SALUD,...,TIPOALTA,MEDICOINTERV1_ENCRIPTADO,ESPECIALIDADINTERVENCION,MEDICOALTA_ENCRIPTADO,USOSPABELLON,IR_29301_COD_GRD,IR_29301_PESO,IR_29301_SEVERIDAD,IR_29301_MORTALIDAD,HOSPPROCEDENCIA
9998,102100,11208,HOMBRE,1950-07-13,NINGUNO,IQUIQUE,IQUIQUE,CHILE,FONASA INSTITUCIONAL - (MAI) B,IQUIQUE,...,DOMICILIO,10736.0,UROLOGÍA,2314.0,1.0,122120,"0,511",0,0,NaN


In [59]:
df_idx = df_sin_fechas.set_index("ID_PACIENTE")
df_idx.loc["11208"]


COD_HOSPITAL                                                           102100
SEXO                                                                   HOMBRE
FECHA_NACIMIENTO                                          1950-07-13 00:00:00
ETNIA                                                                 NINGUNO
PROVINCIA                                                             IQUIQUE
COMUNA                                                                IQUIQUE
NACIONALIDAD                                                            CHILE
PREVISION                                      FONASA INSTITUCIONAL - (MAI) B
SERVICIO_SALUD                                                        IQUIQUE
TIPO_PROCEDENCIA            CENTRO ESPECIALIDADES (CDT, CRS, CONSULTORIO A...
TIPO_INGRESO                                                       PROGRAMADA
ESPECIALIDAD_MEDICA                                                 UROLOGÍA
TIPO_ACTIVIDAD                               CIRUGÍA MAYOR AMBU

<h3> 4. Eliminar duplicados y documentar el proceso.</h3>

In [47]:
dup_ids = df_sin_fechas["ID_PACIENTE"].duplicated().sum()
dup_ids


np.int64(154)

In [48]:
n_ids_repetidos = (df_sin_fechas["ID_PACIENTE"].value_counts() > 1).sum()
n_ids_repetidos


np.int64(96)

In [51]:
ids_repetidos = df_sin_fechas["ID_PACIENTE"].value_counts()
ids_repetidos = ids_repetidos[ids_repetidos > 1]
ids_repetidos.head(20)


ID_PACIENTE
SIN INFORMACIÓN    46
539387               9
301945               6
1019748              4
919522               3
259945               2
562824               2
172835               2
991853               2
720666               2
727230               2
844609               2
486818               2
1165752              2
1378441              2
321631               2
1390930              2
1153428              2
81612                2
212188               2
Name: count, dtype: int64

In [55]:
df_sin_fechas = df_sin_fechas.drop_duplicates(
    subset="ID_PACIENTE",
    keep="first"
)


In [56]:
df_sin_fechas["ID_PACIENTE"].duplicated().sum()

np.int64(0)

In [57]:
df_sin_fechas["ID_PACIENTE"].value_counts()

ID_PACIENTE
11208      1
1314867    1
1354418    1
239861     1
330326     1
          ..
467373     1
1079955    1
674943     1
93557      1
1146635    1
Name: count, Length: 9846, dtype: int64

<h3>5.Documentacion</h5>

<h4 style="color: #2c3e50; border-bottom: 2px solid #3498db; padding-bottom: 5px;">Documentación de Proceso: Limpieza y Preparación de Datos (GRD)</h4>

<p>Esta documentación detalla los pasos realizados en el notebook para procesar el conjunto de datos de Grupos Relacionados por el Diagnóstico (GRD), enfocándose en la reducción de dimensionalidad, normalización de tipos y gestión de registros.</p>
<ol>
        <li>
            <strong>Carga Inicial de Datos</strong>
            <p>El proceso comienza con la importación de la librería <code>pandas</code> y la lectura del archivo fuente (<code>chunk1.csv</code>).</p>
            <ul>
                <li><strong>Configuración:</strong> Se cargan todos los datos como cadenas de texto (<code>dtype=str</code>) para evitar errores de interpretación inicial y se desactiva <code>low_memory</code> para asegurar la consistencia en la lectura.</li>
                <li><strong>Estado Inicial:</strong> El dataset original cuenta con 129 columnas.</li>
            </ul>
        </li>
    </ol>

<div style="font-family: Arial, sans-serif; line-height: 1.6; color: #333;">
    <ol start="2">
        <li>
            <strong style="font-size: 1.1em;">Reducción de Dimensionalidad (Eliminación de Columnas)</strong>
            <p>Se realizó una limpieza profunda eliminando columnas redundantes o con exceso de valores nulos para optimizar el rendimiento del procesamiento:</p>
            <ul>
                <li>
                    <strong>Columnas de Fechas:</strong> Se identificaron todas las columnas que comenzaban con "FECHA". Se conservaron únicamente <code>FECHA_NACIMIENTO</code> y <code>FECHA_INGRESO</code>, eliminando 12 columnas relacionadas con traslados y altas.
                </li>
                <li>
                    <strong>Procedimientos y Diagnósticos:</strong> Se eliminaron bloques masivos de columnas específicas para simplificar la estructura:
                    <ul>
                        <li>30 columnas de "PROCEDIMIENTO" (del 1 al 30).</li>
                        <li>35 columnas de "DIAGNOSTICO" (del 1 al 35).</li>
                    </ul>
                </li>
                <li>
                    <strong>Servicios y Recién Nacidos:</strong> Se eliminaron 9 columnas de <code>SERVICIOTRASLADO</code>, así como todas las columnas relativas a la condición, peso, sexo y estado de neonatos (<code>RN1</code> a <code>RN4</code>).
                </li>
            </ul>
            <p style="margin-top: 10px; padding: 8px; background-color: #f8f9fa; border-left: 5px solid #3498db;">
                <strong>Resultado:</strong> El dataset se redujo de 129 a <strong>27 columnas finales</strong>.
            </p>
        </li>
    </ol>
</div>
<div style="font-family: Arial, sans-serif; line-height: 1.6; color: #333;">
    <ol start="3">
        <li>
            <strong style="font-size: 1.1em;">Normalización y Conversión de Tipos</strong>
            <p>Para habilitar el análisis temporal y cálculos de series de tiempo, se transformaron las columnas críticas de texto a formato de fecha:</p>
            <ul>
                <li><strong>Limpieza de Strings:</strong> Se eliminaron espacios excedentes y se normalizaron los separadores (reemplazando "." y "-" por "/") en <code>FECHA_NACIMIENTO</code> y <code>FECHA_INGRESO</code> para asegurar la compatibilidad con el lector de Pandas.</li>
                <li><strong>Conversión a Datetime:</strong> Se utilizó la función <code>pd.to_datetime</code> con el parámetro <code>errors="coerce"</code> para gestionar fechas inválidas, transformando el tipo de dato de <code>object</code> a <code>datetime64[ns]</code>.</li>
            </ul>
        </li>
<br>
<li>
            <strong style="font-size: 1.1em;">Gestión de Índices e Identificadores</strong>
            <p>Se optimizó la estructura del DataFrame para mejorar la trazabilidad y la velocidad de consulta:</p>
            <ul>
                <li><strong>Renombrado:</strong> La columna original <code>CIP_ENCRIPTADO</code> se renombró a <code>ID_PACIENTE</code> para estandarizar el identificador único.</li>
                <li><strong>Indexación:</strong> Se aplicó el método <code>.set_index("ID_PACIENTE")</code>, lo que permite realizar búsquedas directas y eficientes de registros específicos mediante <code>.loc</code>.</li>
                <li><strong>Reversibilidad:</strong> Se validó el uso de <code>.reset_index()</code> como mecanismo para restaurar el índice numérico secuencial en caso de requerir re-procesamiento.</li>
            </ul>
        </li>

<br>
<li>
            <strong style="font-size: 1.1em;">Tratamiento de Registros Duplicados</strong>
            <p>Se ejecutó un análisis de integridad para garantizar que cada paciente represente una entrada única en el análisis final:</p>
            <ul>
                <li><strong>Identificación:</strong> Se detectaron <strong>154 registros duplicados</strong>. Se identificaron casos con múltiples entradas para un mismo ID (hasta 9 repeticiones) y registros etiquetados como "SIN INFORMACIÓN".</li>
                <li><strong>Depuración:</strong> Se eliminaron las redundancias utilizando <code>drop_duplicates</code>, conservando únicamente la primera aparición (<code>keep="first"</code>) de cada identificador.</li>
            </ul>
            <p style="margin-top: 10px; padding: 10px; background-color: #e8f4fd; border-left: 5px solid #2980b9;">
                <strong>Validación Final:</strong> Se confirmó un total de <strong>0 duplicados</strong> remanentes, consolidando un dataset de <strong>9,846 registros únicos</strong>.
            </p>
        </li>
    </ol>
</div>
